docs/concepts/serialization.md

```diff
@@ -465,6 +484,153 @@ annotated as `<SomeType>`, and type-checkers like mypy will treat the attribute
+
+For example:
+
+```py
+from pydantic import BaseModel
+
+
+class User(BaseModel):
+    name: str
+
+
+class UserLogin(User):
+    password: str
+
+
+class OuterModel(BaseModel):
+    user1: User
+    user2: User
+
+
+user = UserLogin(name='pydantic', password='password')
+
+print(OuterModel(user1=user, user2=user).model_dump(serialize_as_any=True)) # (1)!
+"""
+{
+    'user1': {'name': 'pydantic', 'password': 'password'},
+    'user2': {'name': 'pydantic', 'password': 'password'},
+}
+"""
+
+print(OuterModel(user1=user, user2=user).model_dump(serialize_as_any=False)) # (2)!
+"""
+{
+    'user1': {'name': 'pydantic'},
+    'user2': {'name': 'pydantic'},
+}
+"""
+```
+
+1. With `serialize_as_any` set to `True`, the result matches that of V1.
+2. With `serialize_as_any` set to `False` (the V2 default), fields present on the subclass,
+but not the base class, are not included in serialization.
+
+This setting even takes effect with nested and recursive patterns as well. For example:
+
+```py
+from typing import List
+
+from pydantic import BaseModel
+
+
+class User(BaseModel):
+    name: str
+    friends: List['User']
+
+class UserLogin(User):
+    password: str
+
+
+class OuterModel(BaseModel):
+    user: User
+
+user = UserLogin(name='samuel', password='pydantic-pw', friends=[UserLogin(name='sebastian', password='fastapi-pw', friends=[])])
+
+print(OuterModel(user=user).model_dump(serialize_as_any=True)) # (1)!
+"""
+{
+    'user': {
+        'name': 'samuel',
+        'friends': [
+            {
+                'name': 'sebastian',
+                'friends': [],
+                'password': 'fastapi-pw',
+            }
+        ],
+        'password': 'pydantic-pw',
+    }
+}
+"""
+
+print(OuterModel(user=user).model_dump(serialize_as_any=False)) # (2)!
+"""
+{
+    'user': {
+        'name': 'samuel',
+        'friends': [
+            {
+                'name': 'sebastian',
+                'friends': [],
+            }
+        ],
+    }
+}
+"""
+```
+
+1. Even nested `User` model instances are dumped with fields unique to `User` subclasses.
+2. Even nested `User` model instances are dumped without fields unique to `User` subclasses.
+
+#### Overridng the `serialize_as_any` default (False)
+
+You can override the default setting for `serialize_as_any` at the project level by configuring a subclass of `BaseModel` that overrides the default for the `serialize_as_any` argument to `model_dump()` and `model_dump_json()`.
+
+For example, you could do the following if you want to use duck-typing serialization by default:
+
+```py
+from typing import Any, Dict
+
+from pydantic import BaseModel, SecretStr
+
+class MyBaseModel(BaseModel):
+    def model_dump(self, **kwargs) -> Dict[str, Any]::
+        return super().model_dump(serialize_as_any=True, **kwargs)
+
+    def model_dump_json(self, **kwargs) -> str:
+        return super().model_dump_json(serialize_as_any=True, **kwargs)
+
+
+class User(BaseModel):
```

Comment:
With my suggestion above, it's less important to change this, but either way I think it might make sense to have this be `class User(MyBaseModel):` just to emphasize that if you want that default everywhere, you'll want to have all your models inherit from the custom BaseModel subclass.